source/MaterialXCore/Element.cpp

```diff
@@ -81,6 +81,69 @@ bool Element::operator!=(const Element& rhs) const
     return !(*this == rhs);
 }
 
+bool Element::isEquivalent(ConstElementPtr rhs, ElementEquivalenceOptions& options) const
+{
+    if (getCategory() != rhs->getCategory() ||
+        getName() != rhs->getName())
+    {
+        options.status = "Name or category does not match for elements: '" + getNamePath() + "', '" + rhs->getNamePath() + "'";
+        return false;
+    }
+
+    // Compare attribute names. 
+    StringVec attributeNames = getAttributeNames();
+    StringVec rhsAttributeNames = rhs->getAttributeNames();
+
+    // Filter out any attributes specified in the options.
+    const StringSet& skipAttributes = options.skipAttributes;
+    if (!skipAttributes.empty())
+    {
+        attributeNames.erase(std::remove_if(attributeNames.begin(), attributeNames.end(),
+            [&skipAttributes](const string& attr) { return skipAttributes.find(attr) != skipAttributes.end(); }),
+            attributeNames.end());
+        rhsAttributeNames.erase(std::remove_if(rhsAttributeNames.begin(), rhsAttributeNames.end(),
+            [&skipAttributes](const string& attr) { return skipAttributes.find(attr) != skipAttributes.end(); }),
+            rhsAttributeNames.end());
+    }    
+
+    // Ignore ordering if option specified.
+    if (options.ignoreAttributeOrder)
+    {
+        // Sort the string arrays
+        std::sort(attributeNames.begin(), attributeNames.end());
+        std::sort(rhsAttributeNames.begin(), rhsAttributeNames.end());
+    }
+    if (attributeNames != rhsAttributeNames)
+    {
+        options.status = "Attribute names differ for elements: '" + getNamePath() + "', '" + rhs->getNamePath() + "'";
+        return false;
+    }
+
+    for (const string& attr : rhsAttributeNames)
+    {
+        if (getAttribute(attr) != rhs->getAttribute(attr))
+        {
+            options.status = "Attribute " + attr + " differs for elements: '" + getNamePath() + "', '" + rhs->getNamePath() + "'";
+            return false;
+        }
+    }
+
+    // Compare children.
+    const vector<ElementPtr>& c1 = getChildren();
```

Comment:
I would like to add an option to ignore child-order for `GraphElements` (nodegraphs / documents) only but want to get this up for initial review first.